python/alibiexplainer/alibiexplainer/explainer.py

```diff
@@ -70,7 +71,8 @@ def _predict_fn(self, arr: Union[np.ndarray, List]) -> np.ndarray:
                 instances.append(req_data.tolist())
             else:
                 instances.append(req_data)
-        resp = self.predict({"instances": instances})
+        loop = asyncio.get_running_loop()
+        resp = loop.run_until_complete(self.predict({"instances": instances}))
```

Comment:
@cliveseldon looks like we could not make the explain call to be async as the `predict_fn` is invoked in alibi itself, maybe it is out of the scope of this PR but I think it is still beneficial to make explain call async.

Reply:
Why do we need explain to be async? The explainer will always be a separate Pod that calls the predictor over the network  in batches multiple times until it has enough evidence to make an explanation.

Reply:
The explainer need to run sequential batch requests to the predictor to build up an understanding of the model to create an explanation. It there needs to run those batch requests sequentially. So the explainer predict calls need to be sequential at present I think.

Reply:
The predict function is call here: https://github.com/kubeflow/kfserving/blob/99c604e71fa7355434e13798b0c76bba507e91a2/python/alibiexplainer/alibiexplainer/explainer.py#L73

Could we not simply add an `await` there?

Reply:
Explainers do not need to be async, but the `predict` method in `KFModel` is now async.

My initial thought was to throw in `async` and call it a day as well. You can however not `await` without making the method itself an `async def`. I went down the rabbit hole of trying to make the entire explainer async, but the problem is that a reference to `_predict_fn` is passed to your Alibi lib and is called synchronously there. We would need to refactor the Alibi library to support async, which seems unnecessary.

Awaiting synchronously is what I ended up implementing, that way `_predict_fn` is still synchronous.

I'm personally not super impressed with the async implementation in python. They make it very hard to mix async and sync use cases.